pycgm/static.py

```diff
@@ -0,0 +1,299 @@
+
+@author: cadop
+"""
+import numpy as np
+
+def iad_calculation(rasi, lasi):
+    r"""Inter ASIS Distance (IAD) Calculation function
+
+    Calculates the Inter ASIS Distance from a given frame.
+
+    Given the markers RASI and LASI, the Inter ASIS Distance is
+    defined as:
+
+    .. math::
+
+        InterASISDist = \sqrt{(RASI_x-LASI_x)^2 + (RASI_y-LASI_y)^2 +
+                        (RASI_z-LASI_z)^2}
+
+    where :math:`RASI_x` is the x-coordinate of the RASI marker in frame.
+
+    Parameters
+    ----------
+    rasi : array
+        Array of marker data.
+    lasi : array
+        Array of marker data.
+
+    Returns
+    -------
+    Inter Asis Distance : float
+        The euclidian distance between the two markers.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from .static import iad_calculation
+    >>> rasi = np.array([395.37, 428.1, 1036.83])
+    >>> lasi = np.array([183.19, 422.79, 1033.07])
+    >>> np.around(iad_calculation(rasi, lasi), 2)
+    212.28
+    """
+
+    return np.sqrt(np.sum([(rasi[i] - lasi[i])**2 for i in range(3)]))
+
+def calculate_head_angle(head):
+    r"""Static Head Calculation function
+
+    This function first calculates the x, y, z axes of the head by
+    subtracting the head origin from the given head axes. Then the offset
+    angles are calculated. This is a y-rotation from the rotational matrix
+    created by the matrix multiplication of the distal
+    axis and the inverse of the proximal axis.
+
+    The head axis is calculated as follows:
+
+    ..math::
+
+        head\_axis = \begin{bmatrix}
+            head_{x1}-origin_x & head_{x2}-origin_y & head_{x3}-origin_z \\
+            head_{y1}-origin_x & head_{y2}-origin_y & head_{y3}-origin_z \\
+            head_{z1}-origin_x & head_{z2}-origin_y & head_{z3}-origin_z
+            \end{bmatrix}\\
+
+    The offset angle is defined as:
+
+    .. math::
+
+        \[ result = \arctan{\frac{M[0][2]}{M[2][2]}} \]
+
+    where M is the rotation matrix produced from multiplying axisD and
+    :math:`axisP^{-1}`
+
+    Parameters
+    ----------
+    head : array
+        An array containing the head axis and head origin.
+
+    Returns
+    -------
+    offset : float
+        The head offset angle for static calibration.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from .static import calculate_head_angle
+    >>> head = np.array([[[100.33, 83.39, 1484.08],
+    ...        [98.97, 83.58, 1483.77],
+    ...        [99.35, 82.64, 1484.76]],
+    ...        [99.58, 82.79, 1483.8]], dtype=list)
+    >>> np.around(calculate_head_angle(head), 2)
+    0.28
+    """
+
+    # Calculate head_axis as above [[head_axis_x1 - origin_x, ...], ...]
+    head_axis = np.array([[head[0][y][x] - head[1][x] for x in range(3)]
+                          for y in range(3)])
+
+    # Inversion of [[0, 1, 0], [-1, 0, 0], [0, 0, 1]]
+    inverted_global_axis = [[0, -1, 0], [1, 0, 0], [0, 0, 1]]
```

Comment:
what is this for? I can't remember if it is converting left hand to right hand, or its because it permanently assumes an incorrect global axis.

Reply:
It permanently assumes an incorrect axis.